packages/pages-shared/asset-server/handler.ts

```diff
@@ -582,6 +571,46 @@ export async function generateHandler<
 	}
 }
 
+/**
+ * Given a redirect match and the request URL, returns the response
+ * for the redirect. This function will convert the Location header
+ * to be a root-relative path URL if the request origin and destination
+ * origins are the same. This is so that if the project is served
+ * on multiple domains, the redirects
+ */
+export function getResponseFromMatch(
+	{ status, to }: Pick<MetadataStaticRedirectEntry, "status" | "to">,
```

Comment:
```suggestion
	{ status, to }: Pick<MetadataStaticRedirectEntry | MetadataRedirectEntry, "status" | "to">,
```

Exactly the same shape here once you disregard `lineNumber`, I know, but I could technically be passing in either type here.